### DISCUSSION TITLE
Try code online REPL

### DISCUSSION THREAD

**Discussion by franklupo** [microsoft/fluentui-blazor#2279] [General]
Hi,
I saw the site https://try.fluentui-blazor.net/ it doesn't work. 
When will it be available?

best regards

**Comment by vnbaaij** [comment]
Oh, that has been on the backburner for a while already. Not ready for prime time for a while, if ever. It is a side-project on the side of the side-project (the library) so low priority.

**Reply by franklupo** [reply]
It would make it easier to manage issues with examples.

There would be many examples made by many people for the whole community, available immediately.

With MudBlazor I have used a lot.

**Reply by franklupo** [reply]
IMO it would give a significant boost to the project

**Reply by StevenTCramer** [reply]
@franklupo I think this is where the standard and sincere place for "Pull requests are welcomed" applies :)

**Reply by franklupo** [reply]
Where is the source code of the current REPL?

**Reply by vnbaaij** [reply]
> @franklupo I think this is where the standard and sincere place for "Pull requests are welcomed" applies :)

I dont' think I am that far yet!

**Reply by vnbaaij** [reply]
> Where is the source code of the current REPL?

It is still private. As said it is a side project not part of the library so it is not in (and is not going to be part of) the `microsoft/fluentui-blazor` org/repo

**Reply by vnbaaij** [reply]
I'll see if I can get it running again and then make it public

**Comment by vnbaaij** [comment]
The project is of course based on the original [Blazor REPL](https://github.com/BlazorRepl/BlazorRepl) project (which later became the Telerik Blazor REPL). The MudBlazor REPL is also based on that original source. I am not only changing a lot of the internals, but am also "Fluentizing" the design of course.

I got it compiling again, so here are some teaser screenshots...

Start page:
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/17693e0e-b5d9-4be5-8c7e-e4d7a1325d62)

REPL page: (notice the right pane not having right sizes yet)
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/47d62340-0947-4780-901c-982f00523e66)

Package search (shows installed packages):
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/afe253e9-4364-4c1d-ac28-90397674844f)

Package found with version selector and install option:
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/fda6092d-9ca3-48b7-aecf-b07d678992d7)

Using Message bar when packages install successful/failed:
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/57b7b399-0dc9-44a6-9704-359843913a02)

Showing number of packages installed in AppBar:
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/11e6bda3-8abe-4a91-83b0-b8ef11c746f1)

**Reply by franklupo** [reply]
When do you think it will be ready? I like to try online.

Best regards

**Reply by vnbaaij** [reply]
No idea. In the middle of changing from old project type to .NET 8 Web App. Almost nothing works... 😁
I'll try to publish what I have soon

**Reply by franklupo** [reply]
Can I help you?

**Reply by vnbaaij** [reply]
I'm trying to work on it. But you keep raising issues and discussion that I need to respond to! (😉) 

I'll see if I can get you access.

**Reply by franklupo** [reply]
Excuse me

**Reply by vnbaaij** [reply]
You've been invited.

**Reply by franklupo** [reply]
Thanks
I have already downloaded and I'm already watching 

best reagds

**Comment by vnbaaij** [comment]
Ok, the http://try.fluentui-blazor.net site is sort of running again.

It is not finding all files yet, so you can't compile anything yet. But progress was made!

**Reply by vnbaaij** [reply]
It's working now!
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/c92f0cfa-8b0f-476a-9640-71352152bcde)

**Reply by franklupo** [reply]
Save action not work

**Reply by vnbaaij** [reply]
Correct. I know

**Reply by franklupo** [reply]
I would like to do:
- open issues and use the REPL link
- create categorized examples available with a searchable index page.

**Reply by vnbaaij** [reply]
REPL is not part of this repo and chances are big it will never be. It is **not** Microsoft owned code. Just happens to be worked on by the same person. 

I don't want to promise things about functionalities here. Or give timelines or roadmaps here. It is a separate project.

**Reply by franklupo** [reply]
On saving

![Screenshot_20240714-144112](https://github.com/user-attachments/assets/629f7203-87bb-471b-ae20-940dd15a75f7)
